pkg/engine/harpoon.go

```diff
@@ -234,57 +228,60 @@ func (hc *HarpoonConfig) processRaw(ctx context.Context, repo *api.Target, sched
 				log.Fatalf("%s target file must be of type %s", rawMethod, tag)
 			}
 			path := filepath.Join(directory, fileName)
-			if err := hc.EngineMethod(ctx, path, rawMethod, repo); err != nil {
+			if err := hc.EngineMethod(ctx, path, rawMethod, target); err != nil {
 				log.Fatal(err)
 			}
 		} else {
 			// ... get the files iterator and print the file
 			// .. make sure we're only calling the raw engine method on json files
 			subDirTree.Files().ForEach(func(f *object.File) error {
 				if strings.HasSuffix(f.Name, tag) {
-					path := filepath.Join(directory, repo.Raw.TargetPath, f.Name)
-					if err := hc.EngineMethod(ctx, path, rawMethod, repo); err != nil {
+					path := filepath.Join(directory, target.Raw.TargetPath, f.Name)
+					if err := hc.EngineMethod(ctx, path, rawMethod, target); err != nil {
 						return err
 					}
 				}
 				return nil
 			})
 		}
 	}
-	changes, isDiff := hc.findDiff(gitRepo, directory, rawMethod, repo.Branch, initial)
-	if !isDiff && !initial {
-		klog.Infof("Repo: %s, Method: %s: Nothing to pull.....Requeuing", repo.Name, rawMethod)
+	changes := hc.findDiff(gitRepo, directory, rawMethod, target.Branch)
+	// if first run, don't want to return here
```

Comment:
this note is outdated, need to remove it - I'll let the initial test run finish to be sure things are working, then will push a fix to remove the comment

Reply:
removed it